UpgradedVehicles/QPatch.cs

```diff
@@ -27,8 +30,15 @@ public static void Patch()
                 }
                 //Handle SpeedBooster
                 speedModule = SpeedBooster.GetSpeedBoosterCraftable();
-
-                VehicleUpgrader.SetSpeedBooster(speedModule);
+                VehicleUpgrader.SetNewModule(speedModule);
+                
+                //Handle HullArmorUpgrades (ugly and repetitive...)
+                HullArmorMk2Module = HullArmorMk2.GetHullArmorMk2Craftable();
+                HullArmorMk3Module = HullArmorMk3.GetHullArmorMk3Craftable();
+                HullArmorMk4Module = HullArmorMk4.GetHullArmorMk4Craftable();
+                VehicleUpgrader.SetNewModule(HullArmorMk2Module);
+                VehicleUpgrader.SetNewModule(HullArmorMk3Module);
+                VehicleUpgrader.SetNewModule(HullArmorMk4Module);
```

Comment:
If you find this to be repetitive, you can move the various `VehicleUpgrader.SetNewModule` method calls into the PostPatch event of the corresponding `Craftable`.

Reply:
```CS
// In constructor of Craftable class
{
    OnFinishedPatching += () => { VehicleUpgrader.SetNewModule(this.TechType); };
```